packages/ifpack2/src/Ifpack2_BlockTriDiContainer_def.hpp

```diff
@@ -1239,15 +1243,20 @@ class Impl {
     bool collective_;
     int sweep_step_, bsz_, nvec_;
     std::vector<magnitude_type> wrk_;
+
+    const Teuchos::RCP<const Teuchos::Comm<int> > teuchos_comm_;
```

Comment:
You could just get the Teuchos::Comm from the RowMatrix each time; there's no need to keep it around in the class.

Reply:
@mhoemmen, I chose to impl a NormManager that doesn't know about the RowMatrix; it only knows about the communicator.